packages/glimmer-test-helpers/lib/environment.ts

```diff
@@ -441,14 +452,18 @@ class EmberishCurlyComponentManager implements ComponentManager<EmberishCurlyCom
     return component;
   }
 
-  layoutFor(definition: EmberishCurlyComponentDefinition, component: EmberishCurlyComponent, env: Environment): CompiledBlock {
+  layoutFor(definition: EmberishCurlyComponentDefinition, component: EmberishCurlyComponent, env: TestEnvironment): CompiledBlock {
     let layoutString = definition.layoutString;
 
     if (!layoutString && layoutString !== '') {
       layoutString = component['layout'];
     }
 
-    return compileLayout(new EmberishCurlyComponentLayoutCompiler(layoutString), env);
+    if (env.compiledLayouts[definition.name]) {
+      return env.compiledLayouts[definition.name];
+    }
```

Comment:
Save the result of the lookup in a local variable, and move this to the top (above the layoutString stuff)